### PR TITLE
[UX] Add keyboard swipes

### PR DESCRIPTION
**Pull request by Frenzie**
Only the `e` has a few swipes in this initial commit.

Screenshot of west, northwest, north, northeast, east.
Screenshot_2019-04-08_18-14-05

### CODE REVIEW

File: frontend/ui/data/keyboardlayouts/std.lua [hunk 1]
```diff
@@ -8,7 +17,7 @@ return {
         {  --  1           2       3       4       5       6       7       8       9       10      11      12
             { "Q",        "q",    "„",    "0",    "Й",    "й",    "?",    "!",    "Å",    "å",    "1",    "ª", },
             { "W",        "w",    "!",    "1",    "Ц",    "ц",    "(",    "1",    "Ä",    "ä",    "2",    "º", },
-            { "E",        "e",    "@",    "2",    "У",    "у",    ")",    "2",    "Ö",    "ö",    "3",    "¡", },
+            { "E",          e,    "@",    "2",    "У",    "у",    ")",    "2",    "Ö",    "ö",    "3",    "¡", },
```
Comment (poire-z) [inline frontend/ui/data/keyboardlayouts/std.lua]: or name this local variable (and the future ones) `_e_`, so it's balanced in there, and no risk of confusion (and luacheck complain?) when we use `for k, v in pairs() do` and we implement swipes for the letters k or v ?

File: frontend/ui/widget/virtualkeyboard.lua [hunk 2]
```diff
@@ -311,21 +336,31 @@ function VirtualKeyboard:addKeys()
...
         for j = 1, #self.KEYS[i] do
+            local key
+            local key_chars = self.KEYS[i][j][self.keyboard_layout]
+            if type(key_chars) == "table" then
+                key = key_chars[1]
+            else
+                key = key_chars
+                key_chars = nil
+            end
+            logger.dbg(key)
```
Comment (poire-z) [inline frontend/ui/widget/virtualkeyboard.lua]: Volontary addition, or forgotten `logger.dbg()` ?

Comment (Frenzie) [reply frontend/ui/widget/virtualkeyboard.lua]: Overlooked. :-)

File: frontend/ui/data/keyboardlayouts/std.lua (hunk 1 above)
Comment (Frenzie) [reply frontend/ui/data/keyboardlayouts/std.lua]: I'm not entirely sure what you mean. It's drop-in and all that's needed is keyboard updates. ;-)

(Did you say k? I think `ķ` is about the only one possible. But of course such keys would also be good for punctuation characters.)

Comment (poire-z) [reply frontend/ui/data/keyboardlayouts/std.lua]: I meant just what you just changed, `local _e_ =` on the first line, so nothing else :)
(this `k` thing was just an example of another common single char local variable; `i` in `for i, v in pairs()` would have been another better example, as we could have `ï` `î` :)

**Comment by poire-z** [comment]
(And for a next PR, a little overlay widget that would show onHold, a 3x3 grid with the chars we'd get with a swipe in these 9 directions ? :)

**Comment by Frenzie** [comment]
@poire-z I was actually thinking of ideally doing a little bit more than that, see my latest commit.

```lua
local _e_ = {
    -- default
    "e",
    -- quick swipes
    north = "ë",
    northeast = "é",
    northwest = "è",
    east = "ê",
    west = "ẽ",
    south = "ę",
    southeast = "€",
    southwest = "ė",
    -- extended hold characters, more than 3 probably not obvious
    "ē",
}
```

That could look something like this:

Screenshot_2019-04-09_10-34-43

But GUI is much harder. :-)

**Comment by Biep** [comment]
Nice!

**Pull request merged**